### ISSUE TITLE
Vagrant global-status parsing broken

### ISSUE THREAD

**Issue by danlamanna** [dougm/vagrant-tramp#33]
Here's an example of my `vagrant global-status --machine-readable`:

```
➜  vagrant global-status --machine-readable            
1467524288,,ui,info,id       
1467524288,,ui,info,name      
1467524288,,ui,info,provider   
1467524288,,ui,info,state   
1467524288,,ui,info,directory                           
1467524288,,ui,info,
1467524288,,ui,info,--------------------------------------------------------------------------
1467524288,,ui,info,4f4a0f6  
1467524288,,ui,info,someproject      
1467524288,,ui,info,virtualbox 
1467524288,,ui,info,running 
1467524288,,ui,info,/home/dan/projects/someproject
1467524288,,ui,info,
```

`vagrant-tramp--all-boxes` drops the first [8 lines](https://github.com/dougm/vagrant-tramp/blob/master/vagrant-tramp.el#L59) chopping off the id in my output. Changing it to drop 7 causes everything to start working again.

Curious if this output format is different for other versions/systems, since I've had it work on systems in the past.

Vagrant 1.8.1
Linux system 4.5.7-200.fc23.x86_64 
GNU Emacs 24.5.1 (x86_64-redhat-linux-gnu, GTK+ Version 3.18.9)

**Comment by mspaulding06** [comment]
I've run into the same problem.  The first environment listed from the global-status command is not shown in Emacs.  Changing the drop to 7 fixes the problem.

Vagrant 1.8.4
MacOSX
Emacs 24.5.1

**Comment by shazzner** [comment]
Melpa's version is still broken, mspaulding06 workaround still does the trick.
GNU Emacs 25.1.1 (x86_64-pc-linux-gnu, GTK+ Version 2.24.30)

**Comment by gshimansky** [comment]
I had troubles with vagrant-tramp because it selected a wrong VM out of many VMs that I have. Not sure what purpose `dir_name` variable served, but in my understanding it is redundant. I updated shell script to this version and now it works for me.

```
#!/usr/bin/env bash
### vagrant-tramp-ssh --- connect to the named vagrant box

## Copyright © 2016  The Vagrant-Tramp Contributors

## Author: Ryan Prior <ryanprior@gmail.com>

## This file is not part of GNU Emacs.

## vagrant-tramp is free software: you can redistribute it and/or
## modify it under the terms of the GNU General Public License as
## published by the Free Software Foundation, either version 3 of the
## License, or (at your option) any later version.

## This file is distributed in the hope that it will be useful, but
## WITHOUT ANY WARRANTY; without even the implied warranty of
## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
## General Public License for more details.

## You should have received a copy of the GNU General Public License
## along with this file.  If not, see <http://www.gnu.org/licenses/>.


### Commentary:

## This script is intended for use as a TRAMP connection program when
## establishing a connection to a Vagrant box. See vagrant-tramp.el
## for more details.

read name <<< $(echo "$1" | sed 's/vagrant--//')

if [[ ! "$name" ]]; then name="default"; fi

read id dir <<<\
     $(vagrant global-status --machine-readable \
           | awk -v n=1 '{if(NR>n) print a[NR%n]; a[NR%n]=$0}' \
           | tail -n+8 \
           | cut -d, -f5- \
           | awk -v RS="" \
                 -v name="$name" \
                 "\$2==name && \$5~dir { print \$1; print \$5 }")

cd "$dir"
vagrant ssh "$id"
```

**Comment by max-arnold** [comment]
I did my own fixes for vagrant 2.2.6, MacOS Mojave and Parallels. The `vagrant-tramp-ssh` script looks like this:

```shell
#!/usr/bin/env bash
read -d "\n" dir_name name <<< $(echo "$1" | sed $'s/--/\\\n/')

if [[ ! "$name" ]]; then name="default"; fi

read -d "\n" id dir <<<\
     $(vagrant global-status --machine-readable \
           | sed -e "1,/------/d" -e '$ d' \
           | cut -d, -f5- \
           | awk -v RS="" \
                 -v name="$name" \
                 -v dir="$dir_name" \
                 "\$2==name && \$5~dir { print \$1; print \$5 }")

cd "$dir"
vagrant ssh "$id"
```

I'm not sure though why this script is required in the first place, because it is possible to get ssh connection options via `vagrant ssh-config VMNAME`.

Another fix is to replace the `vagrant-tramp--all-boxes` function:

```lisp
(defun vagrant-tramp--all-boxes ()
  "List of VMs per `vagrant global-status` as alists."
  (let* ((status-cmd "vagrant global-status --machine-readable")
         (status-raw (shell-command-to-string status-cmd))
         (status-lines (split-string status-raw "\n"))
         (status-lines-dropped (-drop-last 3 (-drop (1+ (cl-position "------" status-lines :test 'cl-search)) status-lines)))
         (status-data-raw (--map (mapconcat 'identity
                                            (-drop 4 (split-string it ",")) ",")
                                 status-lines-dropped))
         (status-data (--map (replace-regexp-in-string " " "" it) status-data-raw))
         (status-groups (-split-on "" status-data))
         (vm-attrs '(id name provider state dir)))
    (--map (-zip vm-attrs it) status-groups)))
```
